### PR TITLE
Feature - Invitation actions

### PR DESCRIPTION
**Pull request by joefitter**
* Added resend cancel/delete options to invitations
* Added success notifications

waiting for https://github.com/UKHomeOffice/asl-resolver/pull/118
and
https://github.com/UKHomeOffice/asl-public-api/pull/196

### CODE REVIEW

File: pages/profile/invitations/index.js [hunk 1]
```diff
@@ -25,5 +28,27 @@ module.exports = settings => {
...
+    if (!allowedActions.includes(req.params.action)) {
+      return next(new BadRequestError('Unrecognised action'));
+    }
+
+    const { invitationId, action } = req.params;
+    const isDeletion = action === 'delete';
+    const params = {
+      method: isDeletion ? 'DELETE' : 'PUT'
+    };
+    const suffix = isDeletion ? '' : `/${action}`;
```
Comment (lennym) [inline pages/profile/invitations/index.js]:
```suggestion
    const suffix = isDeletion ? '' : `/delete`;
```

We've just asserted that `action === 'delete'` so might as well be explicit about what's going on.

File: pages/profile/invitations/index.js [hunk 2]
```diff
@@ -1,5 +1,8 @@
+const { merge } = require('lodash');
 const { page } = require('@asl/service/ui');
+const { BadRequestError } = require('@asl/service/errors');
```
Comment (lennym) [inline pages/profile/invitations/index.js]:
```suggestion
const { NotFoundError } = require('@asl/service/errors');
```

File: pages/profile/invitations/index.js [hunk 3]
```diff
@@ -25,5 +28,27 @@ module.exports = settings => {
     }
   })({ schema }));
 
+  app.post('/:invitationId/:action', (req, res, next) => {
+    const allowedActions = ['delete', 'resend', 'cancel'];
+
+    if (!allowedActions.includes(req.params.action)) {
+      return next(new BadRequestError('Unrecognised action'));
```
Comment (lennym) [inline pages/profile/invitations/index.js]:
```suggestion
      return next(new NotFoundError());
```

File: pages/profile/invitations/index.js (hunk 1 above)
Comment (joefitter) [reply pages/profile/invitations/index.js]: that doesn't make any sense :(

Comment (joefitter) [reply pages/profile/invitations/index.js]: if its a DELETE we _dont_ want a suffix

Comment (joefitter) [reply pages/profile/invitations/index.js]: we just DELETE /:id

Comment (lennym) [reply pages/profile/invitations/index.js]: My bad.

**Pull request merged**